### PR TITLE
refactor: clean TS errors, add TS rules, cmn types

### PR DESCRIPTION
**Pull request by ChristopherGabba**
**Summary**

This is my first PR to an open source project ever, so feel free to shoot down. 

1. Added descriptions to some of the types
2. Created a common type between them in some instances
3. Added some additional TSConfig rules used in some big open source projects
4. Removed all the Typescript warnings within useShareIntent.

Trying to keep this PR small and management because I'm still learning. Feel free to shut down this PR.

**Todo**

I added some TODO text to the types that eventually would be ideal to clean up and get one common type. I'm not bold enough to mess with the native code yet.

**Issue**

No Github issues. Just trying to add some comments and be helpful.

### CODE REVIEW

File: src/ExpoShareIntentModule.types.ts [hunk 1]
```diff
@@ -5,55 +5,96 @@ export type ChangeEventPayload = {
...
-
 export interface IosShareIntentFile {
-  path: string;
-  type: string;
+  fileSize?: number; //TODO: Consolidate with size from ShareIntentFile
   fileName: string;
   mimeType: string;
-  fileSize?: number;
+  path: string; //TODO: Consolidate all 3 into filePath/path
+  type: string; //TODO: Eliminate? Duplicate with mimeType
```
Comment (achorein) [inline src/ExpoShareIntentModule.types.ts:88]: is not the same thing, ios module send a generic type as a number (see `IOS_SHARE_TYPE_MAPPING`)

File: src/ExpoShareIntentModule.types.ts [hunk 2]
```diff
@@ -5,55 +5,96 @@ export type ChangeEventPayload = {
...
-};
-
 export interface IosShareIntentFile {
-  path: string;
-  type: string;
+  fileSize?: number; //TODO: Consolidate with size from ShareIntentFile
   fileName: string;
   mimeType: string;
-  fileSize?: number;
+  path: string; //TODO: Consolidate all 3 into filePath/path
```
Comment (achorein) [inline src/ExpoShareIntentModule.types.ts:87]: we don't want to consolidate at this moment, `IosShareIntentFile` is a raw type for native output. consolidation is in `ShareIntentFile`

File: src/ExpoShareIntentModule.types.ts [hunk 3]
```diff
@@ -5,55 +5,96 @@ export type ChangeEventPayload = {
...
-  filePath: string;
   mimeType: string;
-  fileSize?: string;
-  contentUri: string;
 }
 
-export type NativeShareIntent = IosShareIntent | AndroidShareIntent;
-export type NativeShareIntentFile = IosShareIntentFile | AndroidShareIntentFile;
+export type NativeShareIntent = AndroidShareIntent | IosShareIntent
+export type NativeShareIntentFiles = AndroidShareIntentFile | IosShareIntentFile
```
Comment (achorein) [inline src/ExpoShareIntentModule.types.ts:100]:
```suggestion
export type NativeShareIntentFile = AndroidShareIntentFile | IosShareIntentFile
```
its a type not a list

File: src/useShareIntent.tsx [hunk 4]
```diff
@@ -29,38 +28,45 @@ export const SHAREINTENT_OPTIONS_DEFAULT: ShareIntentOptions = {
   disabled: false,
 };
 
-const IOS_SHARE_TYPE_MAPPING = {
```
Comment (achorein) [inline src/useShareIntent.tsx:32]: 🤔 maybe will be useful in the futur

**Comment by achorein** [comment]
@ChristopherGabba  I updated this PR, is that fine for you ?

**Review by ChristopherGabba — COMMENTED** [review]
Thanks @achorein, I appreciate you giving me a few wins on my first commit ever, looks good. :)

**Pull request merged**